concepts/classes/introduction.md

```diff
@@ -0,0 +1,99 @@
+This value will then become the default for newly created objects from this class.
+
+## Encapsulation and Information Hiding
+
+Classes offer the option to restrict access to their members.
+The two basic cases are `private` and `public`.
+`private` members are not accessible from outside the class.
+`public` members can be called freely.
+All members are `private` by default and need to be explicitly marked to be usable outside of the class.
+
+## Basic example
+
+The definition of a new `class` can be seen in the following example.
+Notice the `;` after the definition.
+
+```cpp
+class Wizzard {
+  public:
+    int cast_spell() {
+      return damage;
+    }
+    std::string name{};
+  private:
+    int damage{5};
+};
+
+```
+
+You can access all variables from within the class.
+Take a look at `damage` inside the `cast_spell` function.
+You cannot read or change `private`  members outside of the class:
+
+```cpp
+Wizzard silverhand{};
+// calling the `cast_spell` function is okay, it is public:
+iggwilv.cast_spell();
+// => 5
+// name is public and can be changed:
+silverhand.name = "Laeral";
+// damage is private -> Compilation error
+silverhand.damage = 500; 
+```
+
+## Constructors
+
+Constructors offer the possibility to assign values to member variables at object creation.
+They have the same name as the `class` and do not have return type.
+A class can have several constructors.
+This is useful if you are not always in need to set all variables.
+
+```cpp
+class Wizzard {
+  public:
+    Wizzard(std::string new_name) {
+      name = new_name;
+    }
+    Wizzard(std::string new_name, int new_damage) {
+      name = new_name;
+      damage = new_damage;
+    }
+    int cast_spell() {
+      return damage;
+    }
+    std::string name{};
+  private:
+    int damage{5};
+};
+
+Wizzard el{"Eleven"};       // deals  5 damage
+Wizzard vecna{"Vecna", 50}; // deals 50 damage
+```
+
+Constructors are a big topic and have many nuances.
+If you are not explicitly defining a `constructor` for your `class`, then - and only then - the compiler will do the job for you.
+This has happened in the first example above.
+The _silverhand_ object is created by calling the default constructor, no arguments were passed.
+All variables are set to the value that was stated in the definition of the class.
+If you had not given any values in that definition, the variables would be uninitialized, which might have unintended consequences.
+
+## Structs
+
+Structs are a holdover from the C past that developed into C++.
+They are effectively the same thing as `classes` with one important exception.
+By default, everything in a `class` is `private`.
+Structs, on the other hand are `public` until defined otherwise.
+Conventionally, the `struct` keyword is often used for **data-only structures**.
+The `class` keyword is preferred for objects that have both data **and** function members.
+The class keyword is preferred for objects that have both data and functions.
```

Comment:
I think either line 98 or line 99 .. but not both.

Reply:
@vaeng Oh, I completely missed these lines.  
Are you sure about them? In my experience it's completely fine for a `struct` to have member variables and member functions.

From a technical standpoint a `struct` and a `class` are almost identical, the only differences are that the implicit "default" access specifier (`public` for `struct`, `private` for `class`), and an implicit "default" access-specifier when defining a derive class (`public` inheritance when deriving from a `struct`, `private` inheritance when deriving from a `class`).

So it's "just" about conventions. The advice I heard from leading experts is that classes are for types with an invariant, something that must be true over the lifetime of an object. The constructor is responsible for establishing that invariant and it must hold at the end of each public member function. That's when you need to make member variable `private`, it allows the class to control the values and preserve the invariant. structs are for types without an invariant, where member variables can be modified from the outside. But that doesn't necessarily prevent them for having member functions.

Reply:
You are right. My "convention" is just the way I practiced it in the past.
The "use structs for pod" is surely a consequence of something without an invariant. I wonder how we can write that advice without being too technical.
I went through a lot of math courses and know people who did them with me who never became confident in finding invariants for their logical reasoning. 
Maybe I should give an example like "a wizard's life or mana cannot be negative, so these values are controlled by functions, rather than offer the possibility to set those directly". That might be a bit too wordy? Should this be moved the about.md?

Reply:
What do you think of this change? @siebenschlaefer @BethanyG

Reply:
Looks good!